cloudflare_exporter/collector.py

```diff
@@ -96,3 +127,57 @@ def _get_cloudflare_analytics(token):
                 continue
             raise e
         yield zone['name'], series
+
+
+def nanos2s(time):
+    """Transform nanoseconds to seconds
+
+    :param time: Time in Nanoseconds
+    :return: Time in seconds
+    """
+    return time / 1_000_000
+
+
+class log_metrics:
+    def __init__(self):
+        self.received_requests_pop_origin = Counter()
+        self.origin_response_times = defaultdict(list)
+
+    def add(self, zone, serie):
+        # Log by originIP are useless in Cache Mode
+        if not serie['CacheTieredFill']:
+            key = (zone, serie['EdgeColoCode'], serie['OriginIP'], serie['ClientCountry'])
+            self.received_requests_pop_origin[key] += 1
+            self.origin_response_times[key].append(nanos2s(serie['OriginResponseTime']))
+
+
+def _get_cloudflare_metrics_from_logs(token):
+    cloudflare = CloudFlare.CloudFlare(debug=False, token=token)
+    # from cf docs: Must be at least 1 minute earlier than now and later than start
+    end = int(time.time() - 60 * 1)
+    # We get the last 5 minutes.
+    start = end - 60 * 5
+    for zone in cloudflare.zones.get():
+        series = cloudflare.zones.logs.received(zone['id'],
+                                                params={'end': end,
+                                                        'start': start,
+                                                        'timestamps': 'unixnano',
```

Comment:
Nothing else available? _unix_ or _rfc3339_ to avoid nanos2s then and map this easily to a datetime?

Reply:
> Sounds to me that we're doing a log parser or a log2metrics solution here, is it scalable?
Do you know good lib for that ?
We just want some sampling here, it's enough. We limit the number of line read, no scalling problem.
> Won't it hurt a lot other/existing metrics processing?
I don't understand your question

Reply:
> 
> 
> Nothing else available? _unix_ or _rfc3339_ to avoid nanos2s then and map this easily to a datetime?

the OriginResponseTime is allways in nano... ( even If I set timestamps to _unix_ or _rfc3339_ )
then I remove this line

Reply:
* _rfc3339_: my point was this one:
```
datetime.datetime.fromisoformat('2017-05-08T17:30:34.009056+05:30')
>>> datetime.datetime(2017, 5, 8, 17, 30, 34, 9056, tzinfo=datetime.timezone(datetime.timedelta(seconds=19800)))
```
We could avoid writing our own `nano2s` conversion method.

* About scalability, I asked about the performance of this operation of log parsing. What will happen when the API will return thousands/millions of logs and MBs of output? What will happen if this happens when the scraping from prometheus is in progress? Does this should be done in a background task? etc etc.

Reply:
There is a limit -> 'count': LOGS_EXTENDED_METRIC_COUNT,  
We just need a sample, we don't need all the log to make those stats

Reply:
I receive from clouflare
OriginResponseTime=136000000  #136ms
do you have an easier method to change is to second ?

Reply:
OK, my understanding from https://developers.cloudflare.com/logs/logpull-api/requesting-logs/, was that they unified timestamps by using date everywhere. Got it now, thanks!